ServerCore/Pages/Events/FastestSolves.cshtml.cs

```diff
@@ -45,21 +45,22 @@ public async Task OnGetAsync(SortOrder? sort, PuzzleStateFilter? stateFilter)
 
             DateTime submissionEnd = Event.AnswerSubmissionEnd;
             // get the page data: puzzle, solve count, top three fastest
-            var puzzlesData = await PuzzleStateHelper.GetSparseQuery(_context, this.Event, null, null)
+            // todo: This pulls all solved puzzles for the event. Is that performance acceptable?
+            var puzzlesData = (await PuzzleStateHelper.GetSparseQuery(_context, this.Event, null, null)
                 .Where(s => (s.SolvedTime != null &&
                              s.Puzzle.IsPuzzle &&
                              s.UnlockedTime != null &&
                              s.SolvedTime <= submissionEnd &&
-                             s.Team.IsDisqualified == false))
+                             s.Team.IsDisqualified == false)).ToListAsync())
```

Comment:
What are the perf implications of converting to a list in mid-query rather than at the end? Looks hazardous but I don't really understand the optimization engine.

Reply:
This was probably what was happening before -- EF has never been able to translate this kind of query, so at some point, it would have pulled data locally. That said, this page has also always been slow, so I've made an attempt to actually compute this in the db.